[docs] Add a guide on using icon libraries with Joy UI (#35377)
mui/material-ui: docs/data/joy/guides/using-icon-libraries/using-icon-libraries.md

### Installation

This section assumes that you've already installed Joy UI in your app—see [Installation](/joy-ui/getting-started/installation/) for instructions.

:::warning
To use `@mui/icons-material` with Joy UI, you will have to install Material UI (`@mui/material`) because it is a required dependency. The increase of the production bundle size is very little if you configure [tree shaking](https://developer.mozilla.org/en-US/docs/Glossary/Tree_shaking) in your build process.

We are aware of this limitation and considering removing the dependency. You can keep track of the progress from [this issue](https://github.com/mui/material-ui/issues/34489).
:::

#### npm

```sh
npm install @mui/icons-material @mui/material
```

#### yarn

```sh
yarn add @mui/icons-material @mui/material
```

### Usage

By default, Joy UI components are able to control the color, font-size, and margin of the icons when their size, or variant change.

{{"demo": "JoyMaterialIcon.js"}}

### Size

To control the size of the icon, use `fontSize` prop. The value can be one of the keys in `theme.fontSize` (the default value is `"xl"`).

{{"demo": "IconFontSizes.js"}}

## 3rd-party icons

> To use icon libraries, web font icons, or plain svg icons with Joy UI, apply the styles with specific CSS variables like the example below.

Reviewer: ```suggestion
To use other icon libraries, web font icons, or plain SVG icons with Joy UI, apply the styles with specific CSS variables as shown in the example below:
```

```jsx
import { CssVarsProvider } from '@mui/joy/styles';
import GlobalStyles from '@mui/joy/GlobalStyles';
// The `GlobalStyles` component is used to create a global style sheet.
// You can replace it with your preferred CSS solution.

function App() {
  return (
    <CssVarsProvider>
      <GlobalStyles styles={{
        // The {selector} is the CSS selector to target the icon.

>         // We recommend to use a class over a tag if it is possible.

Reviewer: ```suggestion
        // We recommend using a class over a tag if possible.
```

        '{selector}': {
          color: "var(--Icon-color)",
          margin: "var(--Icon-margin)",
          fontSize: "var(--Icon-fontSize, 20px)",
          width: "1em",
          height: "1em"
        }
      }}>
    </CssVarsProvider>
  )
}
```

Joy UI components can control those variables based on their size and variant to make the icons fit perfectly.

---

Here is a collection of well-known icon libraries that you can use with Joy UI.

> ### React icons

Reviewer: ```suggestion
### React Icons
```

<iframe src="https://codesandbox.io/embed/joy-ui-react-icons-n6jljq?fontsize=12&hidenavigation=1&module=%2Fdemo.tsx&theme=dark"
     style="width:100%; height:250px; border:0; border-radius: 4px; overflow:hidden;"
     title="joy-ui-react-icons"
     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
   ></iframe>

### Ionicons

- [Browse icons](https://ionic.io/ionicons)
- [Installation](https://ionic.io/ionicons/usage)

<iframe src="https://codesandbox.io/embed/inspiring-visvesvaraya-etcc3x?fontsize=12&hidenavigation=1&module=%2Fdemo.tsx&theme=dark"
     style="width:100%; height:250px; border:0; border-radius: 4px; overflow:hidden;"
     title="inspiring-visvesvaraya-etcc3x"
     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
   ></iframe>

### Heroicons

> - [Browse icons](hhttps://heroicons.com/)

Reviewer: ```suggestion
- [Browse icons](https://heroicons.com/)
```

- [Installation](https://github.com/tailwindlabs/heroicons#react)

<iframe src="https://codesandbox.io/embed/joy-ui-heroicons-wv2ev1?fontsize=12&hidenavigation=1&module=%2Fdemo.tsx&theme=dark"
     style="width:100%; height:250px; border:0; border-radius: 4px; overflow:hidden;"
     title="joy-ui-heroicons"
     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
   ></iframe>

> ### Bootstrap icons

Reviewer: ```suggestion
### Bootstrap Icons
```

- [Browse icons](https://icons.getbootstrap.com/)
- [Installation](https://icons.getbootstrap.com/#install)

<iframe src="https://codesandbox.io/embed/joy-ui-bootstrap-icons-x8g0cm?fontsize=12&hidenavigation=1&module=%2Fdemo.tsx&theme=dark"
     style="width:100%; height:250px; border:0; border-radius: 4px; overflow:hidden;"
     title="joy-ui-bootstrap"
     allow="accelerometer; ambient-light-sensor; camera; encrypted-media; geolocation; gyroscope; hid; microphone; midi; payment; usb; vr; xr-spatial-tracking"
     sandbox="allow-forms allow-modals allow-popups allow-presentation allow-same-origin allow-scripts"
   ></iframe>

> ### Fontawesome

Reviewer: ```suggestion
### Font Awesome Icons
```

#### yarn

```sh
yarn add @mui/icons-material @mui/material
```

:::warning
Because `@mui/material` is a required dependency of `@mui/icons-material`, you have add a workaround with yarn resolutions:

```json
{
  "dependencies": {
    "@mui/material": "npm:@mui/joy@latest"
  },
  "resolutions": {
    "@mui/material": "npm:@mui/joy@latest"
  }
}
```

After that, run `yarn install` in your terminal.

We are aware of this limitation and are considering removing the dependency.
You can keep track of the progress in [this issue](https://github.com/mui/material-ui/issues/34489).
:::

#### npm

```sh
npm install @mui/icons-material @mui/material
```

:::warning

Because `@mui/material` is a required dependency of `@mui/icons-material`, you have update your bundler's config to add an alias.

Here is an example of how you can do it, if you use [`webpack`](https://webpack.js.org/):

**webpack.config.js**

```diff
 module.exports = {
   //...
+  resolve: {
+    alias: {
+      '@mui/styled-engine': '@mui/styled-engine-sc'
+    },
+  },
 };
```

If you use TypeScript, you will need to update the TSConfig.

**tsconfig.json**

```diff
 {
   "compilerOptions": {
+    "paths": {

> +      "@mui/styled-engine": ["./node_modules/@mui/styled-engine-sc"]

Reviewer 2: Wrong package :)
